receiver/scraperhelper/scrapercontroller.go

```diff
@@ -146,7 +145,7 @@ func (sc *controller) Shutdown(ctx context.Context) error {
 
 	var errs error
 	for _, scraper := range sc.scrapers {
-		errs = multierr.Append(errs, scraper.Shutdown(ctx))
+		errs = errors.Join(errs, scraper.Shutdown(ctx))
```

Comment:
@seankhliao I see you have participated in some discussions related to the `errors.Join` implementation. Would you mind commenting on whether this pattern should be used? Or should we prefer appending to an `[]error` and only then call `errors.Join`?

Reply:
I think Join replaces nicely `multierr.Combine`, but does not have a good equivalent for `Append`.

Reply:
sorry I missed this last week.
I think semantically it would be more appropriate to populate a slice and join that.
On implementation, `errors.Join(slice...)` would inspect (`errors.Is/As`) the errors based on the order they were added, while `err = errors.Join(err, newErr)` would do so in reverse order.

Reply:
Thanks @seankhliao! Could you update the PR based on this,@shivanshu1333?

Reply:
Sure!

Reply:
done, sorry for the delay